app/models/ArrivalState.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+sealed trait ArrivalState {
+  def transition(messageReceived: MessageReceived): ArrivalState
+}
+
+object ArrivalState extends Enumerable.Implicits {
+  /*
+    007 -> Default state would be ArrivalSubmitted
```

Comment:
Worth removing the commented code we can note the xml messages type some where else like confluence ...